File: SalesOrderManagement/SalesOrder.Database/SalesOrderDbContext.cs
using SalesOrder.Model;
using System.Data.Entity;

namespace SalesOrder.DataBase
{
    public class SalesOrderDbContext : DbContext
    {
        public SalesOrderDbContext(string dbName) : base(dbName)
        {
            Database.SetInitializer(new DropCreateDatabaseAlways<SalesOrderDbContext>());
        }

        public SalesOrderDbContext(string dbName, IDatabaseInitializer<SalesOrderDbContext> initializer) : base(dbName)
        {
            Database.SetInitializer(initializer);
        }

        public DbSet<Customer> Customers { get; set; }
        public DbSet<Order> Orders { get; set; }
        public DbSet<OrderLine> OrderLines { get; set; }
        public DbSet<Product> Products { get; set; }
        public DbSet<Address> Addresses { get; set; }
    }
}

File: SalesOrder/SalesOrder.DataBase.VB/SalesOrderDbContext.vb
Imports System.Data.Entity
Imports SalesOrder.Model

Public Class SalesOrderDbContext
    Inherits DbContext

    Public Sub New(dbName As String, initializer As IDatabaseInitializer(Of SalesOrderDbContext))
        MyBase.New(dbName)
        System.Data.Entity.Database.SetInitializer(initializer)
    End Sub

    Public Customers As DbSet(Of Customer)
    Public Orders As DbSet(Of Order)
    Public OrderLines As DbSet(Of OrderLine)
    Public Products As DbSet(Of Product)
    Public Addresses As DbSet(Of Address)

End Class
